datacatalog/snippets/src/main/java/com/example/datacatalog/CreateCustomConnector.java

```diff
@@ -0,0 +1,257 @@
+    // but does not have to be.
+    String storageProjectId = "my-storage-project";
+
+    // Use any available Dataplex Catalog region.
+    String location = "us-central1";
+
+    /* Use Spark context if you would like to run a connector on GCP as a Dataplex task.
+    At the end of the application, stop the context.
+    JavaSparkContext ctx = new JavaSparkContext(new SparkConf());
+    < rest of the connector code.. >
+    ctx.stop();
+    */
+
+    importEntriesViaCustomConnector(location, projectId, entryGroupId, storageProjectId,
+        gcsBucketName);
+
+
+  }
+
+  public static void importEntriesViaCustomConnector(String location, String projectId,
+      String entryGroupId, String storageProjectId, String gcsBucketName)
+      throws IOException, ExecutionException, InterruptedException {
+
+    // Showing how to fetch metadata from a source system is out of the scope of this sample.
+    // Comments in the method below provide some hints though.
+    fetchMetadataFromSourceSystem();
+
+    // Translate fetched metadata into Dataplex Entry format.
+    DumpItem dumpItem = prepareDumpItem();
+
+    // Write metadata in Dataplex format to an existing Google Cloud Storage bucket.
+    writeMetadataToGscBucket(dumpItem, storageProjectId, gcsBucketName);
+
+    // Call DataplexCatalog ImportEntries() API to import the dump.
+    importEntriesToCatalog(projectId, location, entryGroupId, gcsBucketName);
+
+  }
+
+  private static void fetchMetadataFromSourceSystem() {
+    /* Here is a general approach on example of MySQL database:
+
+    String mySqlUrl = getArg("mysql_url", args);
+    String mySqlUsername = getArg("mysql_username", args);
+    // Don't really pass password as and argument,
+    // use [Secret Manager](https://cloud.google.com/secret-manager) to keep the password safe.
+    String mySqlPassword = getArg("mysql_password", args);
+
+    Class.forName ("com.mysql.jdbc.Driver").newInstance ();
+    Connection conn = DriverManager.getConnection (mySqlUrl, mySqlUsername, mySqlPassword);
+    PreparedStatement ps = conn.prepareStatement(
+        "SELECT table_schema, table_name, create_time, update_time FROM information_schema.tables");
+    ResultSet rs = ps.executeQuery();
+      while (rs.next()) {
+
+      // add Entry basing on ResultSet to some buffer
+      // ...
+    }
+      rs.close();
+      conn.close();
+
+     */
+  }
+
+  private static DumpItem prepareDumpItem() {
+    // Prepare Dataplex Entry based on metadata fetched form the source system.
+
+    Schema schema = Schema.newBuilder()
+        .addColumns(ColumnSchema.newBuilder().setColumn("ID").setType("LONGINT"))
+        .addColumns(ColumnSchema.newBuilder().setColumn("NAME").setType("VARCHAR(20)"))
+        .build();
+    SystemTimestamps timestamps = SystemTimestamps.newBuilder()
+        .setCreateTime(Timestamps.fromDate(new Date()))
+        .setUpdateTime(Timestamps.fromDate(new Date()))
```

Comment:
may be it is better to specify that udpate time and create time should be taken from the source ? To avoid unnecessary update

Reply:
great point